src/SearchParameters/index.js

```diff
@@ -333,12 +342,68 @@ function SearchParameters(newParameters) {
 
   forOwn(params, function checkForUnknownParameter(paramValue, paramName) {
     if (!this.hasOwnProperty(paramName)) {
-      console.error('Unsupported SearchParameter: `' + paramName + '` (this will throw in the next version)');
+      /*eslint-disable*/
```

Comment:
Warnings are to be avoided the same way the errors are! We ought to remove the warning through the rules or in the code. I did it in the code.